src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -21,39 +62,542 @@ const ActorInit En_Daiku_InitVars = {
+    this->subCamAtTarget.z = this->actor.posRot.pos.z;
+
+    Math_SmoothScaleMaxMinF(&this->subCamAt.x, this->subCamAtTarget.x, 1.0f, 1000.0f, 0.0f);
+    Math_SmoothScaleMaxMinF(&this->subCamAt.y, this->subCamAtTarget.y, 1.0f, 1000.0f, 0.0f);
+    Math_SmoothScaleMaxMinF(&this->subCamAt.z, this->subCamAtTarget.z, 1.0f, 1000.0f, 0.0f);
+
+    func_800C04D8(globalCtx, this->subCamId, &this->subCamAt, &this->subCamEye);
+}
+
+void EnDaiku_EscapeSuccess(EnDaiku* this, GlobalContext* globalCtx) {
+    static Vec3f D_809E4148 = { 0, 0, 120 };
+    s32 pad;
+    Vec3f vec;
+
+    Gameplay_ClearCamera(globalCtx, this->subCamId);
+    Gameplay_ChangeCameraStatus(globalCtx, 0, 7);
+    this->subCamActive = false;
+
+    if ((gSaveContext.eventChkInf[9] & 0xF) == 0xF) {
+        Matrix_RotateY(this->initRot.y * (M_PI / 0x8000), MTXMODE_NEW);
+        Matrix_MultVec3f(&D_809E4148, &vec);
+        if (Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_GE3, this->initPos.x + vec.x, this->initPos.y + vec.y,
+                        this->initPos.z + vec.z, 0, Math_atan2f(-vec.x, -vec.z) * (0x8000 / M_PI), 0, 2) == NULL) {
+            Actor_Kill(&this->actor);
+        }
+    } else {
+        func_8002DF54(globalCtx, &this->actor, 7);
+    }
+}
+
+/**
+ * The carpenter is running away
+ */
+void EnDaiku_EscapeRun(EnDaiku* this, GlobalContext* globalCtx) {
+    s32 pad1;
+    Path* path;
+    s16 ry;
+    f32 dx;
+    f32 dz;
+    s32 pad2;
+    f32 dxz;
+    Vec3s* pointPos;
+
+    path = &globalCtx->setupPathList[this->actor.params >> 4 & 0xF];
+    pointPos = (Vec3s*)SEGMENTED_TO_VIRTUAL(path->points) + this->waypoint;
+    dx = pointPos->x - this->actor.posRot.pos.x;
+    dz = pointPos->z - this->actor.posRot.pos.z;
+    ry = Math_atan2f(dx, dz) * (0x8000 / M_PI);
+    dxz = sqrtf(SQ(dx) + SQ(dz));
+    if (dxz <= 20.88f) {
+        this->waypoint++;
+        if (this->waypoint >= path->count) {
+            if (this->subCamActive) {
+                EnDaiku_EscapeSuccess(this, globalCtx);
+            }
+            Actor_Kill(&this->actor);
+            return;
+        }
+    }
+
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, ry, 1, 0xFA0, 0);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    Math_SmoothScaleMaxMinF(&this->actor.speedXZ, this->runSpeed, 0.6f, dxz, 0.0f);
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+
+    if (this->subCamActive) {
+        EnDaiku_UpdateSubCamera(this, globalCtx);
+        if (this->escapeSubCamTimer-- <= 0) {
+            EnDaiku_EscapeSuccess(this, globalCtx);
+        }
+    }
+    
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnDaiku_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
+    s32 curFrame;
+    Player* player = PLAYER;
+
+    if (this->currentAnimIndex == ENDAIKU_ANIM_RUN) {
+        curFrame = (s32)this->skelAnime.animCurrentFrame;
```

Comment:
```suggestion
        curFrame = this->skelAnime.animCurrentFrame;
```

Reply:
this is a f32 to s32 cast and I like to make such downgrading casts explicit, should I really remove the cast?
part of skelanime code eg `SkelAnime_ChangeLinkAnim` does it too

Reply:
Explicit casts should be avoided imo. We don't have any control over all the casts they did so if we start doing explicit casts everywhere it will be a nightmare